tests/acceptance/pagenavigator.js

```diff
@@ -22,8 +22,7 @@ class PageNavigator {
    * @param {string} selector The CSS selector to click on
    */
   async click (selector) {
-    // await this._page.click(selector);
-    await this._page.evaluate((selector) => document.querySelector(selector).click(), selector);
+    await this._page.click(selector);
     await waitTillHTMLRendered(this._page);
```

Comment:
There are other built-in waiting mechanisms for Puppeteer, such as `waitFor`, `waitForSelector`, and `waitForFunction`. Have we tried all these options? I would prefer we use something built-in, rather than roll our own.

Reply:
`waitFor` would require each navigation/click to wait for a fixed time, which may not accurately wait till things are loaded and cause tests to have longer wait time in general. I tried waitForSelector, but it doesn't work super well. Other puppeteer-extra plugins doesn't fit our use case here. I think for now, we should keep this wait till render function

Reply:
The tricky part about `waitForSelector` or `waitForFunction` is having a reliable way of indicating that the page is done loading. We could have the SDK put something in the SDK or storage that we could use to indicate that all of the rendering is done, but that's potentially out of scope here. The waitTillHTMLRendered function is a pretty robust way of waiting for rendering to stop without requiring internal SDK changes